### PR TITLE
Use `pyarrow.fs.HadoopFileSystem` in pyarrow > 2.0.0

### LINKED ISSUE #8213: [BUG] FutureWarning: pyarrow.hdfs.connect is deprecated
### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

2.2.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS 7
- **Python version**: 3.9.6
- **pyarrow version**: 5.0.0

### Describe the problem

When loading a model from the MLflow Model Registry (backend store is HDFS) I get the following warning:

```
legobricks/lib/python3.9/site-packages/mlflow/store/artifact/hdfs_artifact_repo.py:188: FutureWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.
  connected = pa.hdfs.connect(
```

### Steps to reproduce the bug

1. Create a conda env with mlflow and pyarrow installed
2. Start Mlflow with hdfs as backend store
3. Train a model and save it
4. Store the trained model in the model registry
5. Load model from the registry

### Code to generate data required to reproduce the bug

_No response_

### Is the console panel in DevTools showing errors relevant to the bug?

_No response_

### Does the network panel in DevTools contain failed requests relevant to the bug?

_No response_

### PR DESCRIPTION
**Pull request by maksboyarin**
[Open in GitHub Codespaces](https://codespaces.new/maksboyarin/mlflow/pull/9878?quickstart=1)

### Related Issues/PRs

Resolve #8213

### What changes are proposed in this pull request?
Fixed a deprecation warning: 
FutureWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
@maksboyarin thanks for the PR :) 
Could you make this change version-dependent (conditional API usage based on the installed version of PyArrow) and remove the minimum required version pin so that this doesn't break existing environments that have PyArrow pinned to 1.x versions?

File: mlflow/store/artifact/hdfs_artifact_repo.py [hunk 1]
```diff
@@ -183,14 +183,22 @@ def hdfs_system(scheme, host, port):
...
         host = "default"
-
-    connected = pa.hdfs.connect(
-        host=host,
-        port=port or 0,
-        user=kerberos_user,
-        kerb_ticket=kerb_ticket,
-        extra_conf=extra_conf,
-    )
+    if pyarrow.__version__ < "2.0.0":
```
Comment (harupy) [inline mlflow/store/artifact/hdfs_artifact_repo.py:186]: can we use `packaging.version.Version` here?

Comment (harupy) [reply mlflow/store/artifact/hdfs_artifact_repo.py:186]: /ai suggest code

Comment (harupy) [reply mlflow/store/artifact/hdfs_artifact_repo.py:186]: @maksboyarin can you apply the suggestion?

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**